pkg/bundle/parameters.go

```diff
@@ -89,6 +93,7 @@ func (pd ParameterDefinition) CoerceValue(value interface{}) interface{} {
 		if ok {
 			i, ok := asInt(f)
 			if !ok {
+				// Why do we return a float here?
```

Comment:
Reading the `asInt` method, if the two values are not equal, the returned `i` here is 0.
I'd ask @itowlson about the logic, and if returning a float doesn't break  the validation later on.

Reply:
I believe the goal here is to coerce the read value to an integer.  If the value of `f` is integral, then `asInt` returns the corresponding integer (and true).  If `asInt` returns false, then `f` _isn't_ integral, so we can't coerce it to an integer type, but we have to return something so we return `f`.  The comment implies this should never happen because the value should already have been validated as being integral - this is solely a runtime type coercion, not a validation/conversion.

Reply:
Coming back to it at this remove, it is certainly not very self-documenting and could probably be improved with the use of error values or whatever.  And moar comments.

Reply:
LOL... I forgot that I left the comment in there. I'll rewrite this comment and merge something less weird.